hal/timer/driver.go

```diff
@@ -4,61 +4,32 @@
 
 package timer
 
-import (
-	"github.com/embeddedgo/kendryte/p/bus"
-	"github.com/embeddedgo/kendryte/p/timer"
-)
-
-type Driver struct {
-	p *Periph
-
-	channel int
-}
-
-func NewDriver(p *Periph, ch int) *Driver {
-	d := Driver{
-		p:       p,
-		channel: ch,
-	}
-
-	return &d
+type DriverPWM struct {
```

Comment:
Consider the PWM name for this struct. Timers are special case and it's not a real timer driver but rather a higher level interface to one of its channel functionality.